Sockets/BackEndMonitoringSockets/src/CAcceptor.cpp

```diff
@@ -40,32 +40,30 @@ bool CAcceptor::Initialize(const std::string& ip_address,
 	return m_is_acceptor_initialized;
 }
 
-bool CAcceptor::Accept(int& connected_socket_fd)
+bool CAcceptor::AcceptNewClient(int& socket_fd)
 {
-	connected_socket_fd = c_error_socket;
+	int connected_socket_fd = c_error_socket;
 
 	if (m_is_acceptor_initialized)
 	{
-		while (!m_event.WaitFor(std::chrono::nanoseconds(1000)))
+		if (m_is_socket_blocked)
 		{
-			if (m_is_socket_blocked)
-			{
-				connected_socket_fd = AcceptBlockingSockets();
-			}
-			else
-			{
-				connected_socket_fd = AcceptNonBlockingSockets();
-			}
-
-			if (connected_socket_fd > 0)
-			{
-				CLOG_DEBUG_WITH_PARAMS("Acctpted socket ", connected_socket_fd);
-				return true;
-			}
-			else if (connected_socket_fd == c_error_socket)
-			{
-				return false;
-			}
+			connected_socket_fd = AcceptBlockingSockets();
+		}
+		else
+		{
+			connected_socket_fd = AcceptNonBlockingSockets();
+		}
+
+		if (connected_socket_fd > 0)
+		{
+			CLOG_DEBUG_WITH_PARAMS("Accepted socket", connected_socket_fd);
+			socket_fd = connected_socket_fd;
+			return true;
+		}
+		else if (connected_socket_fd == c_error_socket)
+		{
+			return false;
```

Comment:
Log an error?

Reply:
I didn't add logging errors in all places to easily find the main problems. But now I add logging to all classes